server/pubsub/pub.go

```diff
@@ -19,71 +19,53 @@ import (
 	"sync"
 )
 
-type subscriber struct {
-	receiver Receiver
+// Message defines a published message.
+type Message struct {
+	// ID identifies this message.
+	ID string `json:"id,omitempty"`
+
+	// Data is the actual data in the entry.
+	Data []byte `json:"data"`
+
+	// Labels represents the key-value pairs the entry is labeled with.
+	Labels map[string]string `json:"labels,omitempty"`
 }
 
-type publisher struct {
+// Receiver receives published messages.
+type Receiver func(Message)
+
+type Publisher struct {
 	sync.Mutex
 
-	topics map[string]*topic
+	done chan struct{}
+	subs map[*Receiver]struct{}
```

Comment:
Is using an empty struct a common way for this type of map? I saw it done with a `bool` a few times.